### ISSUE TITLE
[FEATURE] Run with given regressor instead of raising warning in ZeroInflatedRegressor

### ISSUE THREAD

**Issue by DoDzilla-ai** [koaning/scikit-lego#480]
I am using the ZeroInflatedRegressor for the prediction of some materials' prices. For some materials I am getting this error:

```
Traceback (most recent call last):
  File "/home/bugra/project-mats/mat_analyser.py", line 61, in apply_regression
    zir.fit(X_train, y_train)
  File "/home/bugra/project-mats/mats_folder/lib/python3.7/site-packages/sklego/meta/zero_inflated_regressor.py", line 107, in fit
    "The predicted training labels are all zero, making the regressor obsolete. Change the classifier or use a plain regressor instead.")
ValueError: The predicted training labels are all zero, making the regressor obsolete. Change the classifier or use a plain regressor instead.
```
Tbh. I didn't understand much from the error. But, I guess using the regressor (in my case this is LinearRegression from sklearn) could give me some results. Why ZeroInflatedRegressor can't use the regressor I defined for it but instead raising this error? I think there should be a flag (a parameter) in ZeroInflatedRegressor which would let it fit the regressor as suggested by the error message if the user wants. If not, it could still raise this error.

**Comment by koaning** [comment]
Just to confirm; can you varify that `y_train` contains no zeros? 

If so, the error makes sense to me. If there are no zeros on the dataset, the classifier in the meta-model can't train.

**Comment by DoDzilla-ai** [comment]
```
zir = ZeroInflatedRegressor(
            classifier=SVC(),
            regressor=LinearRegression()
        )
        try:
            zir.fit(X_train, y_train)
        except ValueError as e:
            if str(e) =="The predicted training labels are all zero, making the regressor obsolete. Change the classifier or use a plain regressor instead.":
                print(y_train.values)
```
And here are some of the y_train values:

```
[1 0 0 1 2 0 0 0 1 0]
[0 1 0 0 0 0 0 1 0 0]
[0 0 0 0 0 1 2 0 0 0]
[3 1 3 0 0 0 0 0 0 0]
[12  0  8  0  0  0  4  0  0  0]
[0 1 2 0 3 0 0 0 0 0]
```

**Comment by MBrouns** [comment]
The thing is that if there are only zero's in your y_train, there is no data left for the regressor to train on. We filter out all zero entries before passing the data to the regressor, as we don't want the zeros to bias the regressors prediction. There's no way the regressor component could still give results in that situation, hence the exception.

**Comment by DoDzilla-ai** [comment]
Exactly @MBrouns . And I think this should be an option given to the user.

**Comment by koaning** [comment]
Ah, you're doing multi-output regression! Now I see. That seems valid and indeed the `ZeroInflatedRegressor` wasn't designed with that in mind. Just looping @Garve in to be sure.

@DoDzilla-ai If we add a flag in the `ZeroInflatedRegressor`, we can use the native [MultiOutputClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.multioutput.MultiOutputClassifier.html) for your use-case, correct?

**Comment by MBrouns** [comment]
I'm thinking something along the lines of `handle_unknown` in sklearns `OneHotEncoder`

```
ZeroInflatedRegressor(handle_zero='error')
ZeroInflatedRegressor(handle_zero='ignore')
```

**Comment by Garve** [comment]
> Ah, you're doing multi-output regression! Now I see. That seems valid and indeed the `ZeroInflatedRegressor` wasn't designed with that in mind. Just looping @Garve in to be sure.

Exactly.

**Comment by koaning** [comment]
@DoDzilla-ai is this an issue you'd like to pick up?

**Comment by DoDzilla-ai** [comment]
@koaning I wish I could, currently struggling with health issues :(

**Comment by koaning** [comment]
No worries, health is more important.

**Comment by mkalimeri** [comment]
Hi all, is someone working on this issue?

**Comment by FBruzzesi** [comment]
> Hi all, is someone working on this issue? 

Hey there! No one is actively working on this! Feel free to pick it up and give it a try. Let us know if you need any help 🙏

**Comment by mkalimeri** [comment]
> > Hi all, is someone working on this issue?
> 
> Hey there! No one is actively working on this! Feel free to pick it up and give it a try. Let us know if you need any help 🙏

Great - I will pick it up!

**Comment by mkalimeri** [comment]
Hi all, I am implementing the update using the handle_error flag as suggested by @MBrouns. I am planning to 
- take the flag handle_error='ignore' into consideration only if there are non non-zero targets and not let it train on the whole dataset (which might contain zero targets). So, when there are **only** zero targets and the user chooses handle_error='ignore', the regressor will train on the whole dataset. 
- Also, I will use the handle_error='error' as default, so the user should actively choose to ignore the fact that the train dataset contains only zero targets. 

Is this the direction you had in mind when implementing the ZeroInflatedRegressor?

**Comment by FBruzzesi** [comment]
Closed by #714
